### ISSUE TITLE
Missing items found when updating Go code

### ISSUE THREAD

**Issue by ethanfrey** [confio/poe-contracts#75]
There was a breaking change in `utils/src/jailing.rs` in how `JailingDuration` was stored. This was not in the Changelog. More than the minor issue of updating the changelog, it shows a tricky case we should ensure our automated tooling detects (check for this in https://github.com/confio/poe-contracts/pull/74)

In valset, `SimulateActiveValidators` is exposed in the production build. This seems too critical a contract to leave such a test entry point in, even if there are auth guards. Let's do this (see original comment: https://github.com/confio/poe-contracts/pull/61#discussion_r797952726)

**Comment by maurolacy** [comment]
> There was a breaking change in `utils/src/jailing.rs` in how `JailingDuration` was stored. This was not in the Changelog.

It is in the CHANGELOG, under the 0.6.0-alpha1 release. It is not flagged as breaking, though.

Now, this probably merits its own issue. `JailingPeriod` and `JailingDuration` are similar but different structs.  `JailingDuration` is being used for the API, whereas `JailingDuration` is being used for both, API and state. This merits some discussion:
```
pub enum JailingDuration {
    Duration(Duration),
    Forever {},
}
```
Where `Duration` in turn is `pub struct Duration(u64)`;

So far, so good. Now `JailingPeriod`:
```
pub enum JailingPeriod {
    Until(Expiration),
    Forever {},
}
```
Where `Expiration` is `pub struct Expiration(Timestamp);`
And `Timestamp` is in turn `pub struct Timestamp(Uint64);`
While `Uint64` is `pub struct Uint64(#[schemars(with = "String")] u64);`

Am I the only one seeing a problem here? That's three (or four, counting the enum) levels of nesting, for in the end passing / storing a `u64` value. I understand each of these wrappers adds functionality, but, is that functionality needed?

And, the two structs. It makes sense that one struct is being used for the API, and the other for the state, but the opposite seems to be the case here: `JailingPeriod` is defined in msg.rs, but used for storing state, and returning values (so, it's part of the API and the state) whereas `JailingDuration` is defined in utils/src/jailing.rs, and used in the API.

Finally, at the API level, it seems clear that passing a wrapped `u64` value (i.e. a `Uint64`) meaning "number of seconds", and with the convention that "zero means forever", would have been simpler.

So, I think further simplification is warranted here.

> More than the minor issue of updating the changelog, it shows a tricky case we should ensure our automated tooling detects (check for this in #74)
> 

Problem with this is, this change impacts the API, but is not directly in the `msg.rs` file. Moreover, the equivalent change in `JailingPeriod`,  which impacts the API, but also storage , is in `msg.rs`, not `state.ts`.

Unless we check the schema diffs directly (for the API), and do a recursive / transitive analysis of the different structs used in `state.rs` (for the state), we wouldn't be able to detect these things through source code diffs.

Perhaps this can be solved through organisation:

- All the API messages must be in `msg.rs`. If they are part of a package, fine, but the source file still needs to be called `msg.rs`.
- All the state structs must be in `state.rs`. If they are part of a package, fine, but the source file still needs to be called `state.rs`.

That means: A struct cannot in principle be used for both API and state. This may seem inconvenient. And in fact, it is. But it has the advantages of: encapsulation, decoupling, and traceability.

Perhaps there are other solutions. We could be generating all the schemas related to state, by example, and diff those.
In the same way we now have a `cargo schema`, we could have a `cargo state`, which generates all the state related files for a package. That would still have to be manually maintained, it would be error prone, etc. But it would be better than the current situation.

> In valset, `SimulateActiveValidators` is exposed in the production build. This seems too critical a contract to leave such a test entry point in, even if there are auth guards. Let's do this (see original comment: [#61 (comment)](https://github.com/confio/poe-contracts/pull/61#discussion_r797952726))

That is being currently addressed in #77.

**Comment by ethanfrey** [comment]
Good comments.

Yes, a deeper simplification would be good

**Comment by ethanfrey** [comment]
Closed by #77
